File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int lt=0, rt=1;
        while(rt<nums.size()){
            while(rt<nums.size() && nums[rt]==nums[rt-1])
                rt++;
            if(rt<nums.size()){
                nums[++lt] = nums[rt];
                rt++;
            }
        }
        return lt+1;
    }
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int n = nums.length, count = 0, lt = 0, rt = 0, prev = -101;
        while (rt < n)
            if (nums[rt] == prev)
                rt++;
            else {
                prev = nums[rt];
                nums[lt++] = nums[rt++];
            }
        return lt;
    }
}